File: linear_programming/cs/lp.cs
using System;
using MathNet.Numerics.LinearAlgebra;
using MathNet.Numerics.Optimization;
using MathNet.Numerics.LinearAlgebra.Double;

class Program
{
    static void Main()
    {
        // Define the objective coefficients
        Vector<double> objectiveCoefficients = DenseVector.OfArray(new double[] { 10.0, 15.0 });

        // Define the constraint matrix
        Matrix<double> constraintMatrix = DenseMatrix.OfArray(new double[,]
        {
            { 2.0, 3.0 },
            { 4.0, 2.0 },
            { 1.0, 0.0 }
        });

        // Define the constraint upper bounds
        Vector<double> constraintBounds = DenseVector.OfArray(new double[] { 100.0, 80.0, 20.0 });

        // Define the constraint types
        Vector<ConstraintType> constraintTypes = DenseVector.OfArray(new ConstraintType[]
        {
            ConstraintType.LessThanOrEqual,
            ConstraintType.LessThanOrEqual,
            ConstraintType.LessThanOrEqual
        });

        // Create a linear programming problem
        var problem = new LinearProgram(objectiveCoefficients, constraintMatrix, constraintBounds, constraintTypes);

        // Solve the linear programming problem
        var solver = new CeresSolver();
        var solution = solver.FindMinimum(problem);

        // Print the results
        Console.WriteLine("Product A units to produce: " + solution.MinimizingPoint[0]);
        Console.WriteLine("Product B units to produce: " + solution.MinimizingPoint[1]);
        Console.WriteLine("Total Profit: $" + (-solution.FunctionInfoAtMinimum.Value));
    }
}

File: linear_programming/matlab/lp.m
% Objective function coefficients
c = [10; 15];

% Constraint matrix A
A = [2, 3; 4, 2; 1, 0];

% Constraint right-hand side
b = [100; 80; 20];

% Lower bounds for variables
lb = [0; 0];

% Upper bounds for variables
ub = [];

% Solve the linear programming problem
options = optimoptions('linprog', 'Algorithm', 'simplex', 'Display', 'off');
[x, fval, exitflag, output] = linprog(-c, A, b, [], [], lb, ub, [], options);

% Calculate the total profit
totalProfit = -fval;

% Print the results
fprintf('Product A units to produce: %.2f\n', x(1));
fprintf('Product B units to produce: %.2f\n', x(2));
fprintf('Total Profit: $%.2f\n', totalProfit);
